Assets/Scripts/Framework/GeoLocation/RouteHighlighter.cs

```diff
@@ -0,0 +1,70 @@
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.Events;
+
+namespace Framework.GeoLocation
+{
+    [RequireComponent(typeof(LineRenderer))]
+    public sealed class RouteHighlighter : MonoBehaviour
+    {
+        private const float INVOKE_DELAY = 0.1f;
+
+        [SerializeField] private UnityEvent withinRange = new UnityEvent();
+        
+        [SerializeField] private List<Transform> routePoints;
+        [SerializeField] private LineRenderer lineRenderer;
+        [SerializeField] private float closeRange = 5f;
+        
+        [SerializeField] private Transform _nextPoint;
+
+        private void Start()
+        {
+            Invoke(nameof(InitLine), INVOKE_DELAY);
+            _nextPoint = routePoints[1];
+            
+        }
+
+        private void Update()
+        {
+            if (Vector3.Distance(transform.position, _nextPoint.position) < closeRange)
+            {
+                withinRange?.Invoke();
+            }
+        }
+
+        public void RemoveLine()
+        {
+            if (routePoints.Count <= 2)
+            {
+                lineRenderer.enabled = false;
+                routePoints.Clear();
+            }
+            else
+            {
+                _nextPoint = routePoints[2];
+                routePoints.RemoveAt(0);
+                lineRenderer.positionCount = 0;
+                InitLine();
+            }
+
+        }
+        private void InitLine()
```

Comment:
Deze naam klopt niet meer met hoe het gebruikt wordt. `Init` is een afkorting van `initialize`, je had het eerst alleen in de `Start` waardoor het klopt. Nu doe je het ook als je dichtbij een point bent. Naam als `UpdateLine` zou beter zijn.
```suggestion
        private void UpdateLine()
```